### PR TITLE
Collapse

### PR DESCRIPTION
**Pull request by thanhtranit**

### CODE REVIEW

File: components/collapse/collapse.component.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  .Class(new _ngxCollapse());
+
+  function _ngxCollapse() {
+    var _ATTRIBUTES = {
+      COLOR: 'color',
+      TYPE: 'type'
+    };
+
+    this.constructor = [
+      ng.platform.browser.BROWSER_APP_PROVIDERS[0][16], // animationBuilder
```
Comment (Minh-Van) [inline components/collapse/collapse.component.js]: Remove unnecessary comment

File: components/collapse/collapse.component.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      ngxBootstrap.ngxCores.ngxRendererService,
+      ng.core.ElementRef,
+      function (animationBuilder, ngxRendererService, elementRef) {
+        this.transitionDuration = 500;
+        this.elementRef = elementRef;
+
+        this.animation = animationBuilder.css();
+        this.animation.setDuration(this.transitionDuration);
+
+        Object.defineProperty(this.__proto__, "collapse", {
```
Comment (Minh-Van) [inline components/collapse/collapse.component.js]: `this.__proto__` is deprecated. Use `Object.getPrototypeOf(this)` for replacement

File: components/collapse/collapse.component.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      this.isCollapse = false;
+      this.isCollapsing = true;
+
+      this.isExpanded = false;
+      this.isCollapsed = true;
+
+      setTimeout(function() {
+          // self.height = '0';
+          // self.isCollapse = true;
+          // self.isCollapsing = false;
```
Comment (Minh-Van) [inline components/collapse/collapse.component.js]: Remove unnecessary comments

File: components/collapse/collapse.component.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+      this.isCollapse = false;
+      this.isCollapsing = true;
+
+      this.isExpanded = true;
+      this.isCollapsed = false;
+
+      this.display = '';
+
+      setTimeout(function() {
```
Comment (Minh-Van) [inline components/collapse/collapse.component.js]: Why did you use setTimeout at here ?

File: components/collapse/collapse.component.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    this.hide = function() {
+      var self = this;
+
+      this.isCollapse = false;
+      this.isCollapsing = true;
+
+      this.isExpanded = false;
+      this.isCollapsed = true;
+
+      setTimeout(function() {
```
Comment (Minh-Van) [inline components/collapse/collapse.component.js]: Why did you use setTimeout at here ?

File: components/collapse/collapse.component.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
+(function (ngxBootstrap) {
+  ngxBootstrap.ngxComponents.ngxCollapseComponent = ng.core.Component({
```
Comment (Minh-Van) [inline components/collapse/collapse.component.js]: This component has lint errors. Please fix it

File: components/collapse/tests/ui/isolated-components/test-case.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+﻿(function (testCases, cores, components, fileService) {
+  testCases.isolatedComponents = ng.core.Component({
+    selector: 'ngx-test-case',
```
Comment (Minh-Van) [inline components/collapse/tests/ui/isolated-components/test-case.js]: This test case has lint errors. Please fix it

File: components/collapse/collapse.component.js (hunk 5 above)
Comment (huuthonguyen76) [inline components/collapse/collapse.component.js]: Why initialize variable `self` and then use `this`?

File: components/collapse/collapse.component.js (hunk 4 above)
Comment (huuthonguyen76) [inline components/collapse/collapse.component.js]: I think this is kind of hack. No one can guarantee it will work properly with settimeout 4.

**Pull request closed without merging**